app/src/app/ApiKeysButton.tsx

```diff
@@ -232,13 +255,13 @@ function ApiKeysDialogContent({
     _prevState: SaveApiKeysState | undefined,
     formData: FormData,
   ) {
-    const result = await saveApiKeys(encryptionKey, formData);
+    const result = await saveApiKeys(
+      encryptionKey,
+      onApiKeysChange,
+      onInvalidateApiKeys,
+      formData,
+    );
     if (!result?.errors) {
-      // Avoid invalidation from causing the closing dialog to suspend
```

Comment:
It seems like storing `Promise.resolve()` in state no longer results in suspense on dialog close.